src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -124,8 +124,44 @@ class Neo4jQueryReadStrategy(filters: Array[Filter] = Array.empty[Filter],
 
     val matchQuery: StatementBuilder.OngoingReadingWithoutWhere = filterRelationship(sourceNode, targetNode, relationship)
 
-    val returning = matchQuery.returning(sourceNode, relationship, targetNode)
-    renderer.render(buildStatement(returning))
+    val aliasedSourceNode = sourceNode.as(Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS)
+    val aliasedTargetNode = targetNode.as(Neo4jUtil.RELATIONSHIP_TARGET_ALIAS)
+    val aliasedRelationship = relationship.getRequiredSymbolicName
+
+    val returnExpressions: Seq[Expression] = if (requiredColumns.isEmpty) {
+      Seq(aliasedRelationship, aliasedSourceNode, aliasedTargetNode)
+    }
+    else {
+      requiredColumns.map(column => {
+        val splatColumn = column.split('.')
+        val entityName = splatColumn.head
+
+        val entity = if (entityName.contains(Neo4jUtil.RELATIONSHIP_ALIAS)) {
```

Comment:
because the name can be `<source>` or `source`. 

Example:
`source.name` or `<source.id>`

Reply:
what happens there is a filed named `sourceOfTruth`?

Reply:
it can't be named like that in the relationship strategy... you just have `source.` `target.` `relationship.`, if you select any other field name it breaks before since it doesn't match the schema.